File: Algorithms/Java/217-contains_duplicate.java
//version 1

class Solution {
    public boolean containsDuplicate(int[] nums) {
    
    HashMap<Integer, Integer> myMap = new HashMap<>();

        for(int i = 0; i < nums.length; i++){

            if(myMap.containsKey(nums[i])){
                return true;
            
            }else{
                myMap.put(nums[i], i);
            }

        }

       return false;

        
    }
}

File: Algorithms/C++/217-contains_duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {

        std::unordered_set<int> mySet;

        for(int n : nums){
            auto temp = mySet.find(n);

            if(temp != mySet.end()){
                return true;
            }

            mySet.insert(n);
        }

        return false;

    }
};